packages/commonwealth/client/scripts/hooks/useRunOnceOnCondition.ts

```diff
@@ -0,0 +1,25 @@
+import { useRef } from 'react';
+import useNecessaryEffect from './useNecessaryEffect';
+
+type UseRunOnceOnConditionProps = {
+  callback: () => void | Promise<() => void>;
+  shouldRun: boolean;
+};
+
+const useRunOnceOnCondition = ({
```

Comment:
what usecase made you to create and use this hook?

Reply:
Sometimes we have to set the component state once, after a react query call finishes. For this, we have been using the `useRef` flag approach. ex:

```
const data = useReactQuery();

useEffect(() => {
    if (!isDataSet.current && data) {   // set component data....   }
},[data])
```

this was getting duplicated in multiple places, so I created the hook which would be simpler to manage

ex:
```
 useRunOnceOnCondition({
    callback: () => {
        // the normal callback we do inside a useEffect
      );
    },
    shouldRun: the_condition_on_which_the_hook_should_run_once,
  });
```